List/LocationList.py

```diff
@@ -138,69 +138,69 @@ def getTreasureList():
+            KH2Treasure(272, "Guardian Soul", LocationTypes=[locationType.OC], InvalidChecks=[itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(513, "Protect Belt (Pain and Panic Cup)", LocationTypes=[locationType.OC, locationType.OCCups], InvalidChecks=[itemType.TROPHY, itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(540, "Serenity Gem (Pain and Panic Cup)", LocationTypes=[locationType.OC, locationType.OCCups], InvalidChecks=[itemType.TROPHY,itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(515, "Rising Dragon (Cerberus Cup)", LocationTypes=[locationType.OC, locationType.OCCups], InvalidChecks=[itemType.TROPHY,itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(542, "Serenity Crystal (Cerberus Cup)", LocationTypes=[locationType.OC, locationType.OCCups], InvalidChecks=[itemType.TROPHY,itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(514, "Genji Shield (Titan Cup)", LocationTypes=[locationType.OC, locationType.OCCups], InvalidChecks=[itemType.TROPHY,itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(541, "Skillful Ring (Titan Cup)", LocationTypes=[locationType.OC, locationType.OCCups], InvalidChecks=[itemType.TROPHY,itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(516, "Fatal Crest (Goddess of Fate Cup)", LocationTypes=[locationType.OC, locationType.OCCups], InvalidChecks=[itemType.TROPHY,itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(517, "Orichalcum+ (Goddess of Fate Cup)", LocationTypes=[locationType.OC, locationType.OCCups], InvalidChecks=[itemType.TROPHY,itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]).setLocationWeight(10),
+            KH2Treasure(518, "Hades Cup Trophy (Paradox Cups)", LocationTypes=[locationType.OC, locationType.OCCups, locationType.OCParadoxCup], InvalidChecks=([] if trueRandom else [itemType.TORN_PAGE, itemType.FIRE, itemType.BLIZZARD, itemType.THUNDER, itemType.CURE, itemType.MAGNET, itemType.REFLECT])+[itemType.TROPHY, itemType.FORM, itemType.SUMMON, itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]),
             KH2Treasure(546, "Zexion (AS) Book of Shadows", LocationTypes=[locationType.OC, locationType.AS], InvalidChecks=[itemType.GROWTH_ABILITY, itemType.SUPPORT_ABILITY, itemType.ACTION_ABILITY]),
```

Comment:
imo Zexion rewards should be late game if the OC2 rewards are considered late game

Reply:
agreed, I missed that.